src/core/storage/sframe_interface/unity_sframe.cpp

```diff
@@ -494,6 +494,10 @@ std::shared_ptr<unity_sframe_base> unity_sframe::select_columns(
   return ret;
 }
 
+std::shared_ptr<unity_sframe_base> unity_sframe::copy(){
+  return select_columns(column_names());
```

Comment:
Instead of this -- which requires computation to make sure the sets of names match up, let's use the copying code at lines 923-925 of this PR: 

```
    auto ret = std::make_shared<unity_sframe>();
    auto new_planner_node = std::make_shared<planner_node>(*(this->get_planner_node()));
    ret->construct_from_planner_node(new_planner_node, this->column_names());
    return ret;
```

This doesn't add any overhead and just creates a new copy by duplicating the planner node.